pulp_rpm/app/migrations/0045_modulemd_fields.py

```diff
@@ -15,8 +16,29 @@
 def populate_new_fields(apps, schema_editor):
     """Populate new fields of modulemd."""
     Modulemd = apps.get_model("rpm", "Modulemd")
+    ModulemdDefaults = apps.get_model("rpm", "ModulemdDefaults")
     modules_to_update = []
 
+    # Fix issue #2786 if already tried to update to 3.18.1
+    modulemds_to_update = []
+    for modulemd in Modulemd.objects.filter(snippet__startswith="b\'"):
+        try:
+            modulemd.snippet = ast.literal_eval(modulemd.snippet).decode("utf8")
+        except (ValueError, TypeError, SyntaxError, MemoryError, RecursionError):
```

Comment:
If it hits one of these we should fail the migration I think.  The transaction should roll back automatically.  

I'm curious where all these exceptions came from?  Was it suggested to handle these by a code snippet or did you run into any of them?

Reply:
In my opinion, if the operation fails, we should clear the snippet and let the user know the best way is to remove a module (or a module default) and re-sync. Otherwise, Pulp will stay in an unusable state.
I didn't run to any of these exceptions. But in the official docs, all of them are listed: https://docs.python.org/3/library/ast.html#ast.literal_eval, so better to catch them (all).

Reply:
On the other hand, wouldn't it mean there was a situation which we didn't anticipate?

Reply:
if anything from those is raised is more situation we cannot expect. As there is anything else what can break the snippet naturally (only manual manipulation with them)

Reply:
But if the only circumstance we can think of where it would fail naturally is manual manipulation, I think it's perfectly fine to fail the migration.  Otherwise, if there is a bug, we need to know about it to fix it, and failing the migration is safer in that respect.